src/aspire/utils/bot_align.py

```diff
@@ -0,0 +1,181 @@
+"""
+This code is an ASPIRE port of Ruiyi Yang's `Bayesian Optimal Transport Alignment`.
+
+`https://github.com/RuiyiYang/BOTalign`
+
+`https://arxiv.org/pdf/2305.12310.pdf`
+"""
+
+import numpy as np
+import pymanopt
+from numpy.linalg import norm
+from scipy.optimize import minimize
+
+from aspire.operators import wemd_embed
+from aspire.utils.rotation import Rotation
+
+
+def align_BO(
+    vol_ref,
+    vol_given,
+    loss_type="wemd",
+    downsampling_level=32,
+    max_iters=200,
+    refine=True,
+    tau=1e-3,
+    man_max_iter=500,
+    man_min_grad=0.1,
+    man_min_sz=0.1,
+    verbosity=0,
+    dtype=None,
+):
+    """
+    This function returns a rotation matrix R that best aligns vol_ref with the rotated version of vol_given.
+
+    :param vol_ref: The reference `Volume`
+    :param vol_given: The `Volume` to be aligned
+    :param loss_type: 'wemd' or 'eu'. Default 'wemd'.
```

Comment:
“eu” most often refers to a certain supranational political union. I think we can afford to spell out “euclidean” here.

Reply:
lol, fair, changed.